### PR TITLE
Backport build yml and publishing script changes

### PR DESCRIPTION
**Pull request by JoeRobich**
Build on Arcade update from https://github.com/dotnet/roslyn/pull/61045

Fix failing builds by back porting the 16.11 build yml and script. - https://dev.azure.com/dnceng/internal/_build/results?buildId=1743258&view=logs&s=6884a131-87da-5381-61f3-d7acc3b91d76&j=20fcf628-b65c-5865-625a-1cef81cda63b (microsoft)

Test Build - https://dev.azure.com/dnceng/internal/_build/results?buildId=1750368&view=results (microsoft)

### CODE REVIEW

File: eng/common/SetupNugetSources.ps1 [hunk 1]
```diff
@@ -146,16 +146,22 @@ $userName = "dn-bot"
 # Insert credential nodes for Maestro's private feeds
 InsertMaestroPrivateFeedCredentials -Sources $sources -Creds $creds -Username $userName -Password $Password
 
-$dotnet3Source = $sources.SelectSingleNode("add[@key='dotnet3']")
```
Comment (dibarbet) [inline eng/common/SetupNugetSources.ps1:149]: should we be modifying these?  since this is the arcade owned bit

**Review by dibarbet — APPROVED** [review]

File: eng/common/SetupNugetSources.ps1 (hunk 1 above)
Comment (JoeRobich) [reply eng/common/SetupNugetSources.ps1:149]: I am pulling in an updated Arcade https://github.com/dotnet/roslyn/pull/61044

Comment (dibarbet) [reply eng/common/SetupNugetSources.ps1:149]: Ah - :shipit:

Comment (JoeRobich) [reply eng/common/SetupNugetSources.ps1]: Changed to build on latest NET3 arcade https://github.com/dotnet/roslyn/pull/61045

**Comment by JoeRobich** [comment]
MacOS failures are related to searching for files outside the working directory. Merging

**Pull request merged**